pkg/splunk/controller/configmap.go

```diff
@@ -130,7 +130,7 @@ func SetConfigMapOwnerRef(ctx context.Context, client splcommon.ControllerClient
 	currentOwnerRef := configMap.GetOwnerReferences()
 	// Check if owner ref exists
 	for i := 0; i < len(currentOwnerRef); i++ {
-		if reflect.DeepEqual(currentOwnerRef[i], splcommon.AsOwner(cr, false)) {
+		if reflect.DeepEqual(currentOwnerRef[i].UID, cr.GetUID()) {
```

Comment:
You mentioned that we can include the Kind as well. In that line of thought, just wondering, is it a good idea, just to exclude the API version alone?  A new API that compares all the fields except API version.
```
`// AsOwner returns an object to use for Kubernetes resource ownership references.
func AsOwner(cr MetaObject, isController bool) metav1.OwnerReference {
	return metav1.OwnerReference{
		APIVersion: cr.GetObjectKind().GroupVersionKind().GroupVersion().String(),
		Kind:       cr.GetObjectKind().GroupVersionKind().Kind,
		Name:       cr.GetObjectMeta().GetName(),
		UID:        cr.GetObjectMeta().GetUID(),
		Controller: &isController,
	}
}`
```

Reply:
we do not need controller, we just need  UID. UID will uniquely identify the resource. please [refer](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#:~:text=A%20Kubernetes%20systems%2Dgenerated%20string,(also%20known%20as%20UUIDs).).

Reply:
Does the UID remain the if we update the name, kind of the CR(I'm not sure if these are even possible)? If yes then comparing the UID's should be enough.